File: VB-Examples/13_Shape/AddShapes/Form1.vb
Imports Spire.Doc
Imports Spire.Doc.Documents
Imports Spire.Doc.Fields

Namespace AddShapes
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            Dim doc As New Document()
            Dim sec As Section = doc.AddSection()
            Dim para As Paragraph = sec.AddParagraph()
            Dim x As Integer = 60, y As Integer = 40, lineCount As Integer = 0
            For i As Integer = 1 To 20
                If lineCount > 0 AndAlso lineCount Mod 8 = 0 Then
                    para.AppendBreak(BreakType.PageBreak)
                    x = 60
                    y = 40
                    lineCount = 0
                End If
                'add shape and set its size and position
                Dim shape As ShapeObject = para.AppendShape(50, 50, CType(i, ShapeType))
                shape.HorizontalOrigin = HorizontalOrigin.Page
                shape.HorizontalPosition = x
                shape.VerticalOrigin = VerticalOrigin.Page
                shape.VerticalPosition = y + 50
                x = x + CInt(shape.Width) + 50
                If i > 0 AndAlso i Mod 5 = 0 Then
                    y = y + CInt(shape.Height) + 120
                    lineCount += 1
                    x = 60
                End If

            Next i
            'Save docx file
            doc.SaveToFile("AddShape.docx", FileFormat.Docx)

            'Launch Word file.
            WordDocViewer("AddShape.docx")
        End Sub

        Private Sub WordDocViewer(ByVal fileName As String)
            Try
                Process.Start(fileName)
            Catch
            End Try
        End Sub
    End Class
End Namespace

File: 13_Shape/AddShapes/Form1.cs
using System;
using System.Windows.Forms;
using Spire.Doc;
using Spire.Doc.Documents;
using Spire.Doc.Fields;

namespace AddShapes
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Create Word document.
            Document doc = new Document();
            Section sec = doc.AddSection();
            Paragraph para = sec.AddParagraph();
            int x = 60, y = 40, lineCount = 0;
            for (int i = 1; i < 20; i++)
            {
                if (lineCount > 0 && lineCount % 8 == 0)
                {
                    para.AppendBreak(BreakType.PageBreak);
                    x = 60;
                    y = 40;
                    lineCount = 0;
                }
                //Add shape and set its size and position.
                ShapeObject shape = para.AppendShape(50, 50, (ShapeType)i);
                shape.HorizontalOrigin = HorizontalOrigin.Page;
                shape.HorizontalPosition = x;
                shape.VerticalOrigin = VerticalOrigin.Page;
                shape.VerticalPosition = y + 50;
                x = x + (int)shape.Width + 50;
                if (i > 0 && i % 5 == 0)
                {
                    y = y + (int)shape.Height + 120;
                    lineCount++;
                    x = 60;
                }

            }
            doc.SaveToFile("AddShape.docx", FileFormat.Docx);

            //Launch Word file.
            WordDocViewer("AddShape.docx");
        }

        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}